packages/plugins/users-permissions/admin/src/hooks/useFetchRole/index.js

```diff
@@ -17,17 +20,21 @@ const useFetchRole = id => {
       });
     }
 
+    return () => (isMounted.current = false);
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [id]);
 
   const fetchRole = async roleId => {
     try {
       const { role } = await request(`/${pluginId}/roles/${roleId}`, { method: 'GET' });
 
-      dispatch({
-        type: 'GET_DATA_SUCCEEDED',
-        role,
-      });
+      // Prevent updating state on an unmounted component
+      if (isMounted.current) {
```

Comment:
When does this occur?

Reply:
You can do this to reproduce it:
- In `users-permissions/admin/src/hooks/useFetchRole/index.js`, remove all references of `isMounted` (comment out lines 9, 12, 23, 32, 37)
- Run the test file `users-permissions/admin/src/pages/Roles/EditPage/tests/index.test.js`
- See the warning "Can't perform a React state update on an unmounted component"

I doesn't make the test fail though, it's just a big red warning

Reply:
And I say run the test file because this issue doesn't seem to occur in a real browser 🤔